File: src/main/java/org/moeftc/ServiceDependencyResolver.java
package org.moeftc;

import org.gradle.api.Project;
import org.gradle.api.artifacts.DependencyResolutionListener;
import org.gradle.api.artifacts.ResolvableDependencies;

public class ServiceDependencyResolver implements DependencyResolutionListener {
    Project project;

    public ServiceDependencyResolver(Project project) {
        this.project = project;
    }



    @Override
    public void beforeResolve(ResolvableDependencies resolvableDependencies) {
        project.getRepositories().maven(mavenArtifactRepository -> mavenArtifactRepository.setUrl("https://jitpack.io"));
        project.getDependencies().add("implementation", Constants.serviceDependency);
        project.getGradle().removeListener(this);
    }

    @Override
    public void afterResolve(ResolvableDependencies resolvableDependencies) {
    }
}

File: src/main/java/org/moeftc/ServiceDependencyResolver.kt
package org.moeftc

import org.gradle.api.Project
import org.gradle.api.artifacts.DependencyResolutionListener
import org.gradle.api.artifacts.ResolvableDependencies
import org.gradle.api.artifacts.repositories.MavenArtifactRepository
import org.gradle.kotlin.dsl.maven

class ServiceDependencyResolver(var project: Project) : DependencyResolutionListener {
    override fun beforeResolve(resolvableDependencies: ResolvableDependencies) {
        project.repositories.mavenLocal()
        project.repositories.maven("https://jitpack.io")
        project.dependencies.add("implementation", Constants.serviceDependency)
        project.gradle.removeListener(this)
    }

    override fun afterResolve(resolvableDependencies: ResolvableDependencies) {}

}